python/ray/_private/runtime_env/packaging.py

```diff
@@ -217,21 +169,19 @@ def parse_uri(pkg_uri: str) -> Tuple[Protocol, str]:
             f"Invalid protocol for runtime_env URI {pkg_uri}. "
             f"Supported protocols: {Protocol._member_names_}. Original error: {e}"
         )
-    if protocol == Protocol.S3 or protocol == Protocol.GS:
-        return (protocol, f"{protocol.value}_{uri.netloc}{uri.path.replace('/', '_')}")
-    elif protocol == Protocol.HTTPS:
-        parsed_netloc = uri.netloc.replace(".", "_").replace(":", "_").replace("@", "_")
-        return (
-            protocol,
-            f"https_{parsed_netloc}{uri.path.replace('/', '_')}",
-        )
-    elif protocol == Protocol.FILE:
-        return (
-            protocol,
-            f"file_{uri.path.replace('/', '_')}",
-        )
-    else:
-        return (protocol, uri.netloc)
+
+    package_name = uri.netloc
+    if protocol in Protocol.remote_protocols():
+        package_name = f"{protocol.value}_{uri.netloc}{uri.path}"
+
+    disallowed_chars = ["/", ":", "@", "+"]
```

Comment:
I just look at the https://stackoverflow.com/questions/1033424/how-to-remove-bad-path-characters-in-python, do we need to add them all? (Although it is not quite ideal too, i thought we can have forbidden chars provided by python directly...)

Reply:
I'd prefer to be a bit conservative here. I picked characters that I've seen in user URIs in the past. We can expand this list if there's new URIs formats that have unwanted characters.